### ISSUE TITLE
Superset 4.0.2: Superset SQL lab throwing error with presto connection.

### ISSUE THREAD

**Issue by rabindragogoi** [apache/superset#30083]
### Bug description

@michael-s-molina : Superset 4.0.2 throws out error while querying with presto connection.

Attached with the screenshot.

### How to reproduce the bug

1. New setup of superset on top of new mariadb metadata database.
2. make a presto connection.
3. query in sql lab using new presto connection made in step-3.

### Screenshots/recordings

![image](https://github.com/user-attachments/assets/99203e49-43d3-4a65-93c6-0ca3fb060397)

### Superset version

master / latest-dev

### Python version

3.11

### Node version

16

### Browser

Chrome

### Checklist

- [X] I have searched Superset docs and Slack and didn't find a solution to my problem.
- [X] I have searched the GitHub issue tracker and didn't find a similar bug report.
- [X] I have checked Superset's logs for errors and if I found a relevant Python stacktrace, I included it here as text in the "additional context" section.

**Comment by rabindragogoi** [comment]
@michael-s-molina : we are not changing anything in the code. Simply we are using pip install for 4.0.2 with added dependent packages.

The presto database connection we created is as usual and when we tried to query in SqlLab, it showed up the error that I posted. This is working perfect in version 3.0.0.

Need your help to get it fix so that we can upgrade our existing Superset version from 3.0.0 to latest 4.0.2. Also any workaround or suggestion is most welcome.

Thanks in advance.
